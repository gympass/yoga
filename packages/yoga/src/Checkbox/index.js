import { Checkbox, Switch } from './web';

const ExportCheckbox = Object.assign(Checkbox, {
  Switch,
});

export default ExportCheckbox;
