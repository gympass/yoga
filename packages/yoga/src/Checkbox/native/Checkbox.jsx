import React, { useState } from 'react';
import { bool, func, string, oneOf, shape } from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { TouchableWithoutFeedback, View } from 'react-native';
import { hexToRgb } from '@gympass/yoga-common';
import { Check } from '@gympass/yoga-icons';

const CheckboxWrapper = styled.View`
  flex-direction: row;
  align-items: center;
`;

const Label = styled.Text(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  padding-left: ${checkbox.label.padding.left}px;

  font-size: ${checkbox.label.font.size}px;

  color: ${checkbox.label.font.color};
`,
);

const HelperWrapper = styled.View(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  margin-top: ${checkbox.helper.margin.top}px;
`,
);

const Helper = styled.Text(
  ({
    error,
    theme: {
      yoga: {
        colors: { negative },
        components: { checkbox },
      },
    },
  }) => `
  font-size: ${checkbox.helper.font.size}px;

  color: ${error ? negative[1] : checkbox.helper.font.color};
`,
);

const CheckBackground = styled.View(
  ({
    checked,
    disabled,
    error,
    variant,
    theme: {
      yoga: {
        colors: { [variant]: color = [], negative },
        components: { checkbox },
      },
    },
  }) => `
    position: absolute;

    width: ${checkbox.size}px;
    height: ${checkbox.size}px;

    border-radius: ${checkbox.border.radius}px;
    border-width: ${checkbox.border.width}px;
    border-style: solid;

    ${checked ? `background-color: ${color[3]};` : ''}

    border-color: ${disabled ? checkbox.disabled.border.color : color[3]};

    ${
      disabled && checked
        ? `background-color: ${checkbox.disabled.backgroundColor};`
        : ''
    }

    ${error ? `border-color: ${negative[1]};` : ''}

    ${error && checked ? `background-color: ${negative[1]};` : ''}
  `,
);

const Shadow = styled.View(
  ({
    variant,
    theme: {
      yoga: {
        colors: { [variant]: color = [] },
        components: { checkbox },
      },
    },
  }) => {
    const size = checkbox.size * 1.67;

    return `
      width: ${size}px;
      height: ${size}px;

      background-color: ${hexToRgb(color[1], 0.75)};          
      border-radius: ${checkbox.hover.border.radius}px;
    `;
  },
);

const CheckArea = styled.View(
  ({
    theme: {
      yoga: {
        components: { checkbox },
      },
    },
  }) => `
  justify-content: center;
  align-items: center;

  width: ${checkbox.size}px;
  height: ${checkbox.size}px;
`,
);

/** The checkbox component is used when the user needs to select one or more
 * items on a task. This component can also allow the user to turn an option on
 * or off.  */
const Checkbox = ({
  label,
  helper,
  disabled,
  checked,
  error,
  style,
  variant,
  onPressIn,
  onPressOut,
  theme: {
    yoga: {
      components: { checkbox },
    },
  },
  ...rest
}) => {
  const [pressed, setPressed] = useState(false);
  return (
    <View>
      <TouchableWithoutFeedback
        {...rest}
        disabled={disabled}
        onPressIn={e => {
          setPressed(true);
          onPressIn(e);
        }}
        onPressOut={e => {
          setPressed(false);
          onPressOut(e);
        }}
      >
        <CheckboxWrapper style={style}>
          <CheckArea>
            {pressed && !disabled && (
              <Shadow pressed={pressed} variant={variant} />
            )}
            <CheckBackground
              {...{
                disabled,
                checked,
                error,
                pressed,
                variant,
              }}
            />
            {checked && (
              <Check
                fill={checkbox.checked.icon.color}
                style={{ position: 'absolute' }}
              />
            )}
          </CheckArea>

          {label && <Label>{label}</Label>}
        </CheckboxWrapper>
      </TouchableWithoutFeedback>
      {(helper || error) && (
        <HelperWrapper>
          <Helper error={error}>{error || helper}</Helper>
        </HelperWrapper>
      )}
    </View>
  );
};

Checkbox.propTypes = {
  label: string,
  /** set a short helper text under checkbox */
  helper: string,
  checked: bool,
  disabled: bool,
  error: string,
  style: shape({}),
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['primary', 'secondary', 'tertiary']),
  onPressIn: func,
  onPressOut: func,
};

Checkbox.defaultProps = {
  label: undefined,
  helper: undefined,
  checked: false,
  disabled: false,
  error: undefined,
  style: {},
  variant: 'primary',
  onPressIn: () => {},
  onPressOut: () => {},
};

Checkbox.displayName = 'Checkbox';

export default withTheme(Checkbox);
