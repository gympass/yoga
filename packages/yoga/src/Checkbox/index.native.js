import { Checkbox, Switch } from './native';

const ExportCheckbox = Object.assign(Checkbox, {
  Switch,
});

export default ExportCheckbox;
