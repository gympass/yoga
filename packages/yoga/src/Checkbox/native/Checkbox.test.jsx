import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Checkbox from '..';
import ThemeProvider from '../../Theme';

const data = {
  label: 'Checkbox Component',
  helper: 'Helper Text',
};

describe('<Checkbox />', () => {
  describe('Snapshots', () => {
    it('should match snapshot without label', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot without label', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with default checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with disabled checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with checked checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} checked />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with disabled and checked checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} disabled checked />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with error checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} error="Error text" />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with error and checked checkbox', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} error="Error text" checked />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with inverted', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} inverted />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with inverted and checked', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} inverted checked />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with inverted and checked and disabled', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Checkbox {...data} inverted checked disabled />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call onChange function when press on Checkbox', () => {
      const onChangeMock = jest.fn();
      const { getByTestId } = render(
        <ThemeProvider>
          <Checkbox {...data} onPress={onChangeMock} testID="checkbox" />
        </ThemeProvider>,
      );

      fireEvent.press(getByTestId('checkbox'));
      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
