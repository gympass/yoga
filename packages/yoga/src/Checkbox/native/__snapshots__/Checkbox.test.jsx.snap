// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> Snapshots should match snapshot with checked checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "#D8385E",
              "borderColor": "#D8385E",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        fill="#FFFFFF"
        focusable={false}
        height={24}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "position": "absolute",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        width={24}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with default checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#D8385E",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled and checked checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={true}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "#D7D7E0",
              "borderColor": "#D7D7E0",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        fill="#FFFFFF"
        focusable={false}
        height={24}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "position": "absolute",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        width={24}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with disabled checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={true}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#D7D7E0",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error and checked checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={false}
        error="Error text"
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "#CA4808",
              "borderColor": "#CA4808",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        fill="#FFFFFF"
        focusable={false}
        height={24}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "position": "absolute",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        width={24}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4294967295,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      error="Error text"
      style={
        Array [
          Object {
            "color": "#CA4808",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Error text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with error checkbox 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        error="Error text"
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#CA4808",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      error="Error text"
      style={
        Array [
          Object {
            "color": "#CA4808",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Error text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with inverted 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#FFFFFF",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with inverted and checked 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={false}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#FFFFFF",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        fill="#D8385E"
        focusable={false}
        height={24}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "position": "absolute",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        width={24}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4292360286,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot with inverted and checked and disabled 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={true}
        disabled={true}
        inverted={true}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "#D7D7E0",
              "borderColor": "#D7D7E0",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
      <RNSVGSvgView
        bbHeight={24}
        bbWidth={24}
        fill="#D8385E"
        focusable={false}
        height={24}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "position": "absolute",
            },
            Object {
              "opacity": 1,
            },
            Object {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        width={24}
      >
        <RNSVGGroup
          fill={
            Array [
              0,
              4292360286,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          font={Object {}}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGSvgView>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "paddingLeft": 8,
          },
        ]
      }
    >
      Checkbox Component
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": 4,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#6B6B78",
            "fontFamily": "Rubik",
            "fontSize": 14,
            "fontWeight": "400",
          },
        ]
      }
    >
      Helper Text
    </Text>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot without label 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#D8385E",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`<Checkbox /> Snapshots should match snapshot without label 2`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
    >
      <View
        checked={false}
        disabled={false}
        inverted={false}
        pressed={false}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#D8385E",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 2,
              "height": 24,
              "position": "absolute",
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
